core/dbt/config/project.py

```diff
@@ -202,6 +202,17 @@ def _raw_project_from(project_root: str) -> Dict[str, Any]:
     return project_dict
 
 
+def _query_comment_from_cfg(
+        cfg_query_comment: Union[QueryComment, str]
+) -> QueryComment:
+    if isinstance(cfg_query_comment, str):
+        if cfg_query_comment in ('None', ''):
```

Comment:
`if not cfg_query_comment` is probably sufficient here, as well, right?

Reply:
`if not cfg_query_comment` check does not catch below config.
```
query-comment: None
```
because `cfg_query_comment`'s type is `str` and its value is `'None'`

Reply:
I don't think we _should_ catch that case! In yaml, the appropriate thing to do to get `None` out is either
```
query-comment:
```
or
```
query-comment: null
```

Reply:
The [docs](https://docs.getdbt.com/docs/building-a-dbt-project/dbt-projects/configuring-query-comments/#configuring-query-comments)  says that 
```
You can disable query comments by setting the query-comment config to None
```

What is the preferred way of disabling query comments?

Reply:
The docs are wrong! I'll open a PR there to fix that, thank you for pointing that out.

The two methods I listed above should both work.

Reply:
Now that I think about it more, this problem is why I had the `NoValue` construct, to differentiate "not set, so leave it at the default" and "don't supply a query comment".

Reply:
So you are suggesting something like this:
```
query_comment: Optional[Union[NoValue, Union[QueryComment, str]]
```
Is my understanding correct?

Reply:
Yeah, though unions get collapsed by `typing`, so that's equivalent to `Optional[Union[NoValue, QueryComment, str]]`